lib/handlers/quote/quote.ts

```diff
@@ -210,13 +212,19 @@ export class QuoteHandler extends APIGLambdaHandler<
       protocols = [Protocol.V3]
     }
 
+    let parsedDebugRoutingConfig = {}
+    if (debugRoutingConfig && unicornSecret === process.env.UNICORN_SECRET) {
```

Comment:
I think we can be more strict - requiring both `unicornSecret` and `process.env.UNICORN_SECRET` to be populated. Because otherwise, when both are undefined, we are also parsing the `debugRoutingConfig`